lib/plugins/villager.js

```diff
@@ -215,24 +199,25 @@ function inject (bot, { version }) {
     }
   }
 
-  async function putRequirements (window, Trade, count) {
+  async function putRequirements (window, Trade) {
     const [slot1, slot2] = window.slots
-    const { stackSize: stackSize1, type: type1, metadata: metadata1 } = Trade.inputItem1
-    const tradeCount1 = Trade.realPrice
-    const neededCount1 = Math.min(stackSize1, tradeCount1 * count)
-
-    const input1 = !slot1
-      ? neededCount1
-      : (slot1.count < tradeCount1 ? neededCount1 - slot1.count : 0)
-    await deposit(window, type1, metadata1, input1, 0)
+    const { type: type1, metadata: metadata1 } = Trade.inputItem1
+
+    const input1 = slot1
+      ? Math.max(0, Trade.realPrice - slot1.count)
+      : Trade.realPrice
```

Comment:
Only put the requirements for a single trade at a time. This seems to greatly increase reliability, especially for trades with multiple inputs.